subprojects/core-api/src/main/java/org/gradle/api/file/FileSystemLayout.java

```diff
@@ -0,0 +1,40 @@
+/*
+ * Copyright 2023 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.gradle.api.file;
+
+import org.gradle.api.Incubating;
+import org.gradle.api.provider.Provider;
+
+import java.io.File;
+
+/**
+ * Provides provider-based access to file system locations.
+ *
+ * @since 8.4
+ */
+@Incubating
+public interface FileSystemLayout {
+    /**
+     * Creates a {@link RegularFile} provider whose location is calculated from the given {@link Provider}.
```

Comment:
❓ What's the behavior when the given `file` contains a relative path?

Reply:
> ❓ What's the behavior when the given `file` contains a relative path?

It resolves the path relative to the settings dir (ou root dir - by default, they are the same, not sure how to customize it).

Reply:
🗒️  It would be good to document that behavior.

Reply:
I think we should definitely improve the `ProjectLayout` documentation, and add clarifications. It would be great, doing it in the same PR.

The settings directory (or at least the location of the settings file) can be currently customized with the `--settings-file` flag. Though, the flag is currently deprecated. We should add a test for this.

The settings directory, IIUC, can be different from the root directory in a composite build, where each build would have its own settings directory, but the root directory would be the same. We should add a test for this too.

Reply:
I added the clarifications to ProjectLayout and SettingsLayout w.r.t. the reference location when resolving relative paths into absolute ones.